features/items/step_definitions.rb

```diff
@@ -73,3 +73,7 @@
   actual = @instance.send(:"#{attribute}")
   expect(actual).to eq(expected)
 end
+
+When(/^we call "([^"]*)" on aws\-record item instance with a value of "([^"]*)"$/) do |method, value|
```

Comment:
I like the genericism - but this would only apply to integers. I also think that "with a value of" should be optional, so that in your test, you can test default and non-default case.

```
When we call "increment_counter!" on aws-record item instance
When we call "increment_counter!" on aws-record item instance with an integer value of "5"
```

Checking if it says "integer value" or "string value" could determine whether to call `to_i` or `to_s`. That also would apply to any string related item tests.

Reply:
Updated step to be optional and handle both cases. I changed the optional part with "with integer value of..." since the value will always be an integer data type for the method to work. My unit tests covers the data type check.